From 7aaf8b8231ea292e2e07bcf8dc8a64ecc3c14e1c Mon Sep 17 00:00:00 2001
From: Yumi Yukimura <me.cafebabe@gmail.com>
Date: Tue, 15 Sep 2026 04:23:21 +0800
Subject: [PATCH] evolution: Opt-out faceunlock

on non-arm64:

TARGET module FaceUnlock requires non-existent TARGET module: libFaceDetectCA
TARGET module FaceUnlock requires non-existent TARGET module: libMegviiUnlock
TARGET module FaceUnlock requires non-existent TARGET module: libMegviiUnlock-jni-1.2
---
 config/evolution.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/config/evolution.mk b/config/evolution.mk
index 972dfd70..828965b7 100644
--- a/config/evolution.mk
+++ b/config/evolution.mk
@@ -57,6 +57,7 @@ ifeq ($(TARGET_BUILD_DEVICE_AS_WEBCAM), true)
 endif
 
 # Face Unlock
+ifneq ($(TARGET_DISABLE_FACEUNLOCK),true)
 ifeq ($(TARGET_SUPPORTS_64_BIT_APPS),true)
 ifeq ($(TARGET_SUPPORTS_GFU),true)
 $(call inherit-product-if-exists, vendor/google/faceunlock/config.mk)
@@ -71,6 +72,7 @@ PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.hardware.biometrics.face.xml:$(TARGET_COPY_OUT_SYSTEM_EXT)/etc/permissions/android.hardware.biometrics.face.xml
 endif
 endif
+endif
 
 # Enable Material Design 3 Expressive
 PRODUCT_PRODUCT_PROPERTIES += is_expressive_design_enabled=true
-- 
2.47.3

